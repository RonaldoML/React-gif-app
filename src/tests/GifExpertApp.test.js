
import React from 'react';
import '@testing-library/jest-dom';
import GifExpert from '../GifExpertApp';
const { shallow } = require("enzyme");


describe('Prueba de GifExpertApp', () => {
    
    

    test('should mostrar el componente correctamente', () => {

        let wrapper = shallow(<GifExpert></GifExpert>);

        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('should mostrar una lista de categorÃ­a', () => {
        
        const categorias = ['One Punch', 'Dragon Ball'];

        let wrapper = shallow(<GifExpert defaultCategories={categorias}></GifExpert>);

        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('GifGrid').length ).toBe( categorias.length );

    })
    

});
