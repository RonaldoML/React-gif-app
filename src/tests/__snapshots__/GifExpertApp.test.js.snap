// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifExpertApp should mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Gif Searcher
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Lol"
      key="Lol"
    />
  </ol>
</Fragment>
`;

exports[`Prueba de GifExpertApp should mostrar una lista de categor√≠a 1`] = `
<Fragment>
  <h2>
    Gif Searcher
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
